### PR TITLE
Add ENABLE_SAGA_TESTS cmake config, default to off, enable in ci

### LINKED ISSUE #32143: TestSagaAlgorithms: line 139, in check_algorithm: 'NoneType' object has no attribute 'checkParameterValues'
I'm guessing lack of some dependency is not disabling Saga test ?
```
======================================================================
ERROR: test_algorithms:1
'Raster tools - Invert grid', {'algorithm': 'saga:invertgrid', 'name': 'Raster  (__main__.TestSagaAlgorithms)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/qgis/qgis-no-override-testdata/python/plugins/processing/tests/AlgorithmsTestBase.py", line 139, in check_algorithm
    ok, msg = alg.checkParameterValues(parameters, context)
AttributeError: 'NoneType' object has no attribute 'checkParameterValues'
```

### PR DESCRIPTION
**Pull request by strk**
Closes #32143

### CODE REVIEW

**Comment by strk** [comment]
I'm not sure default to off is good. I'd actually default to whether SAGA was found on the system

File: python/plugins/processing/tests/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,13 @@ PLUGIN_INSTALL(processing tests ${PY_FILES})
 PLUGIN_INSTALL(processing tests/testdata ${TEST_DATA_FILES})
 
 IF(ENABLE_TESTS)
+
+  # enable postgresql tests
```
Comment (DelazJ) [inline python/plugins/processing/tests/CMakeLists.txt]:
```suggestion
  # enable SAGA tests
```

**Comment by nyalldawson** [comment]
I think it's fine to default to off. Saga is only available on a limited subset of our supported platforms, and I think trying to get too smart here isn't worth the time required.

**Comment by strk** [comment]
Ok, thanks. Will not backport as it's a new "feature".

**Pull request merged**